Source/BugsnagPlatformConditional.h

```diff
@@ -0,0 +1,33 @@
+//
+//  BugsnagPlatformConditional.h
+//  Bugsnag
+//
+//  Created by Jamie Lynch on 01/06/2020.
+//  Copyright © 2020 Bugsnag. All rights reserved.
+//
+
+// ***IMPORTANT NOTE***: this should always be imported as the first header in a file,
+// because it relies on preprocessor macros. If this is not done the targets will
+// not be defined appropriately and the functions/defines will behave unexpectedly.
+
+#ifndef BugsnagPlatformConditional_h
+#define BugsnagPlatformConditional_h
+
+#import <Foundation/Foundation.h>
+
+/**
+ * Defined as true if this is the iOS platform.
+ */
+#define BSG_PLATFORM_IOS TARGET_IPHONE_SIMULATOR || TARGET_OS_IPHONE
```

Comment:
I don't believe we should be using `TARGET_IPHONE_SIMULATOR` unless we're trying to determine the difference between a real device and a simulator. This will be true for mobile, TV or watch simulators.

Reply:
changed the define to use `TARGET_OS_IOS`